### DISCUSSION TITLE
Separate color for border title(s)

### DISCUSSION THREAD

**Discussion by Z1ni** [Textualize/textual#2209] [Ideas]
It would be nice to be able to set the border (sub)title colors separately from the border color.
For example, a border could be colorized with `$primary` and the title(s) could be colorized with `$secondary`.
Maybe this could be `border-title-color: <color>` etc. in the CSS?

**Comment by davep** [comment]
If it helps, titles are processed for Rich markup:

```python
from textual.app        import App, ComposeResult
from textual.containers import Container
from textual.widgets    import Header, Footer

class BorderColourApp( App[ None ] ):

    CSS = """
    Screen {
        align: center middle;
    }
    Container {
        border: round red;
        width: 40%;
        height: 40%;
    }
    """

    def compose( self ) -> ComposeResult:
        yield Header()
        yield Container()
        yield Footer()

    def on_mount( self ):
        self.query_one( Container ).border_title = "[green]This isn't red![/]"
        self.query_one( Container ).border_subtitle = "[blue]Neither is this![/]"

if __name__ == "__main__":
    BorderColourApp().run()
```

![Screenshot 2023-04-03 at 21 18 32](https://user-images.githubusercontent.com/28237/229618292-e28faf3e-e128-441d-8607-7c683f8b1369.png)

**Reply by Z1ni** [reply]
Yeah, I noticed that and am currently using it for my project, but I'm not sure if there's a way to use the themed colors with the Rich markup? (Semi)hard-coding the color by checking the `dark` property isn't as nice as dedicated coloring.

**Reply by davep** [reply]
Aye, that's right; but I thought it worth a mention in case you didn't know. :-)

**Reply by willmcgugan** [reply]
I have been think that the console markup should use the color system definitions. So "[$primary]Hello World" would pick the right color.

But CSS properties for these things also makes sense. Thanks for the suggestions @Z1ni
